### PR TITLE
Alphabet

### PR DESCRIPTION
**Pull request by NatalijaGucevska**

### CODE REVIEW

File: Main.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Main {
+
+	public static void main(String[] args) {
```
Comment (nmeabed) [inline Main.java:6]: Good tests, maybe add a test of a null argument to make it perfect.
BTW if you hadn't try working with JUnit tests yet, I think you should give it a try :)

File: Language.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	 * Topological sort of the vertices of an acyclic graph
+	 * 
+	 * For the language with sorted words [ART, RAT, CAT, CAR] 
+	 * valid sorts are: 
+	 * [A, T, R, C] and [T, A, R, C]
+	 * 
+	 * return - A list of sorted vertices.  
+	 * 
+	 */
+	private List<Character> topologicalSort(Map<Character, Set<Character>> graph, Map<Character, Integer> degree) {
```
Comment (nmeabed) [inline Language.java:147]: I wasn't familiar with this implementation of the topological sort, I only knew the DFS based one, this is good too.

**Review by nmeabed — APPROVED** [review]
I don't have so much to add, but it's a really nice solution :D

File: Language.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+				String adjacentWord = sortedWords.get(i + 1);
+	
+				boolean wait = true;
+				int count = 0;
+				int limit = Math.min(currenttWord.length(), adjacentWord.length());
+				//Don't enter if word is empty
+				wait = wait && count < limit;
+				
+				while (wait) {
+					if (currenttWord.charAt(count) != adjacentWord.charAt((count))) {
```
Comment (Thomas-de-Rivaz) [inline Language.java]: pulling currenttWord.charAt(count) out into a local variable would clean this up a lot.

File: Language.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+						// Make sure that vertices with zero indegree will be in the
+						// degree map
+						degree.putIfAbsent(currenttWord.charAt(count), 0);
+	
+						// Keep track of the indegree of every vertex while
+						// constructing the graph
+						int deg = degree.getOrDefault(adjacentWord.charAt(count), 0);
+						degree.put(adjacentWord.charAt(count), deg + 1);
+	
+						wait = false;
```
Comment (Thomas-de-Rivaz) [inline Language.java]: using break; instead of the wait would be more explicit that you are quitting the while loop.

**Review by Thomas-de-Rivaz — COMMENTED** [review]
Very good solution. Clean and well commented.
